### PR TITLE
Updated PPP, UG and DG

### PR DESCRIPTION
**Pull request by PawanPatil19**
Do not review/merge... Still working on it

### CODE REVIEW

File: docs/images/ModelClassDiagram.png
Comment (Yaladah) [inline docs/images/ModelClassDiagram.png:1]: What's the arrow pointing from ModelManager pointing at?

File: docs/images/TaskModelClassDiagram.png
Comment (Yaladah) [inline docs/images/TaskModelClassDiagram.png:1]: The arrows at the top right are kind of confusing

**Review by Yaladah — COMMENTED** [review]
Check the arrows I mentioned.

**Review by Cyrus-Krispin — APPROVED** [review]
LGTM 👍

**Pull request merged**